devito/petsc/iet/nodes.py

```diff
@@ -1,31 +1,78 @@
-from devito.ir.iet import Expression
-from devito.ir.equations import OpMatVec, OpRHS
+from devito.ir.iet import Expression, Callable, Call
+from devito.ir.equations import OpInjectSolve
+from devito.tools import as_tuple
 
 
 class LinearSolverExpression(Expression):
-
     """
     Base class for general expressions required by a
     matrix-free linear solve of the form Ax=b.
     """
     pass
 
 
-class MatVecAction(LinearSolverExpression):
-
+class InjectSolveDummy(LinearSolverExpression):
     """
-    Expression representing matrix-vector multiplication.
+    Placeholder expression to run the iterative solver.
     """
-
-    def __init__(self, expr, pragmas=None, operation=OpMatVec):
+    def __init__(self, expr, pragmas=None, operation=OpInjectSolve):
         super().__init__(expr, pragmas=pragmas, operation=operation)
 
 
-class RHSLinearSystem(LinearSolverExpression):
+class PETScCallable(Callable):
+    def __init__(self, name, body, retval=None, parameters=None,
+                 prefix=None, unused_parameters=None):
+        super().__init__(name, body, retval, parameters, prefix)
+        self._unused_parameters = as_tuple(unused_parameters)
+
+    @property
+    def unused_parameters(self):
+        return self._unused_parameters
+
+
+class Callback(Call):
```

Comment:
why did you move this here? it's a generic node type, so it belongs to ir/iet/nodes.py